File: Tests/ClaudeUsageWidgetTests/FormattingTests.swift
import XCTest
@testable import ClaudeUsageWidget

final class FormattingTests: XCTestCase {
    func testLevelTiers() {
        XCTAssertEqual(UsageLevel.forPercent(0), .calm)
        XCTAssertEqual(UsageLevel.forPercent(69.9), .calm)
        XCTAssertEqual(UsageLevel.forPercent(70), .warn)
        XCTAssertEqual(UsageLevel.forPercent(89.9), .warn)
        XCTAssertEqual(UsageLevel.forPercent(90), .hot)
        XCTAssertEqual(UsageLevel.forPercent(150), .hot)
    }

    func testResetCountdown() {
        let now = Date(timeIntervalSince1970: 0)
        let in2h47m = Date(timeIntervalSince1970: 2 * 3600 + 47 * 60)
        XCTAssertEqual(Formatting.resetCountdown(to: in2h47m, now: now), "resets in 2h 47m")
        let in40m = Date(timeIntervalSince1970: 40 * 60)
        XCTAssertEqual(Formatting.resetCountdown(to: in40m, now: now), "resets in 40m")
        XCTAssertEqual(Formatting.resetCountdown(to: now, now: now), "resetting now")
        XCTAssertEqual(Formatting.resetCountdown(to: nil, now: now), "")
    }

    func testUpdatedAgo() {
        let now = Date(timeIntervalSince1970: 1000)
        XCTAssertEqual(Formatting.updatedAgo(Date(timeIntervalSince1970: 990), now: now), "updated 10s ago")
        XCTAssertEqual(Formatting.updatedAgo(Date(timeIntervalSince1970: 700), now: now), "updated 5m ago")
        XCTAssertEqual(Formatting.updatedAgo(now, now: now), "updated just now")
    }
}

File: windows/tests/TokenSpendie.Windows.Tests/Util/FormattingTests.cs
using System;
using FluentAssertions;
using TokenSpendie.Windows.Util;
using Xunit;

namespace TokenSpendie.Windows.Tests.Util;

public class FormattingTests
{
    private static readonly DateTimeOffset Now = DateTimeOffset.FromUnixTimeSeconds(1_700_000_000);

    [Fact] public void ResetCountdownNullReturnsEmpty() =>
        Formatting.ResetCountdown(null, Now).Should().Be("");

    [Fact] public void ResetCountdownPastReturnsResettingNow() =>
        Formatting.ResetCountdown(Now.AddSeconds(-5), Now).Should().Be("resetting now");

    [Fact] public void ResetCountdownMinutesOnly() =>
        Formatting.ResetCountdown(Now.AddMinutes(40), Now).Should().Be("resets in 40m");

    [Fact] public void ResetCountdownHoursAndMinutes() =>
        Formatting.ResetCountdown(Now.AddMinutes(2 * 60 + 47), Now).Should().Be("resets in 2h 47m");

    [Fact] public void ResetDateNullReturnsEmpty() =>
        Formatting.ResetDate(null).Should().Be("");

    [Fact] public void ResetDateFormatsAsWeekday()
    {
        var date = new DateTimeOffset(2026, 5, 25, 0, 0, 0, TimeSpan.Zero); // Monday
        Formatting.ResetDate(date).Should().Be("resets Mon, May 25");
    }

    [Fact] public void UpdatedAgoJustNow() =>
        Formatting.UpdatedAgo(Now, Now).Should().Be("updated just now");

    [Fact] public void UpdatedAgoSeconds() =>
        Formatting.UpdatedAgo(Now.AddSeconds(-10), Now).Should().Be("updated 10s ago");

    [Fact] public void UpdatedAgoMinutes() =>
        Formatting.UpdatedAgo(Now.AddMinutes(-5), Now).Should().Be("updated 5m ago");

    [Fact] public void UpdatedAgoHours() =>
        Formatting.UpdatedAgo(Now.AddHours(-2), Now).Should().Be("updated 2h ago");
}
